dockerfiles/docker-compose.yml

```diff
@@ -162,3 +161,24 @@ services:
     networks:
       readthedocs:
     command: ["server", "--console-address", ":9009", "/data"]
+    
+  createbuckets:
```

Comment:
When this `createbuckets` will be executed? Every time the `docker-compose up` is run?

If that's the case, we avoided to do that by using the `INIT` variable, so it's run only the first time the environment is setup (passing `--init`)

Reply:
Should add a check on `$INIT` :+1:  Currently, it just runs the commands exits anyways since the buckets already exist.

Reply:
Added the check on `$INIT`.

For the sake of simplicity, I like keeping this entrypoint in `.yml` instead of mounting a separate .sh file.

Reply:
The entrypoint script got so bloated that it's now back :)

Reply:
If possible, I'd like to avoid "spinning up an extra docker container each time to do nothing". We could execute the script from `web` container instead and check for `INIT` as we are already doing.

I think you mentioned somewhere else that `mc` is not available on `web`. However, it seems the installation of `mc` client is pretty simple, just downloading a binary file: https://docs.min.io/minio/baremetal/reference/minio-mc.html#quickstart

Reply:
However, probably the simplest way to do this is to re-use the container `storage` and use a bash script as entry point:

```
if INIT:
  # create containers
  mc mb ...
  mc policy ...


server --console-address :9009 /data
```

Looks simpler and cleaner. Also avoids spinning a container that does nothing.

Reply:
We opted to use the `web` container to host the `mc` client and this is now added to the PR :+1: